pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -68,81 +61,26 @@ func NewEtcdCertSignerController(
 	eventRecorder events.Recorder,
-) *EtcdCertSignerController {
+) factory.Controller {
 	c := &EtcdCertSignerController{
 		kubeClient:           kubeClient,
 		operatorClient:       operatorClient,
 		infrastructureLister: infrastructureInformer.Lister(),
 		secretLister:         kubeInformers.SecretLister(),
 		nodeLister:           kubeInformers.InformersFor("").Core().V1().Nodes().Lister(),
-
-		secretClient: v1helpers.CachedSecretGetter(kubeClient.CoreV1(), kubeInformers),
-
-		cachesToSync: []cache.InformerSynced{
-			operatorClient.Informer().HasSynced,
-			kubeInformers.InformersFor("").Core().V1().Nodes().Informer().HasSynced,
-			kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Secrets().Informer().HasSynced,
-			infrastructureInformer.Informer().HasSynced,
-		},
-		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "EtcdCertSignerController"),
-		eventRecorder: eventRecorder.WithComponentSuffix("etcd-cert-signer-controller"),
-	}
-
-	kubeInformers.InformersFor("").Core().V1().Nodes().Informer().AddEventHandler(c.eventHandler())
-	kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Secrets().Informer().AddEventHandler(c.eventHandler())
-	kubeInformers.InformersFor(operatorclient.GlobalUserSpecifiedConfigNamespace).Core().V1().Secrets().Informer().AddEventHandler(c.eventHandler())
-	infrastructureInformer.Informer().AddEventHandler(c.eventHandler())
-
-	return c
-}
-
-func (c *EtcdCertSignerController) Run(i int, stopCh <-chan struct{}) {
-	defer utilruntime.HandleCrash()
-	defer c.queue.ShutDown()
-
-	klog.Infof("Starting EtcdCertSignerController")
-	defer klog.Infof("Shutting down EtcdCertSignerController")
-
-	if !cache.WaitForCacheSync(stopCh, c.cachesToSync...) {
-		utilruntime.HandleError(fmt.Errorf("caches did not sync"))
-		return
-	}
-
-	go wait.Until(c.runWorker, time.Second, stopCh)
-
-	go wait.Until(func() {
-		c.queue.Add(workQueueKey)
-	}, time.Minute, stopCh)
-
-	<-stopCh
-}
-
-func (c *EtcdCertSignerController) runWorker() {
-	for c.processNextWorkItem() {
-	}
-}
-
-func (c *EtcdCertSignerController) processNextWorkItem() bool {
-	dsKey, quit := c.queue.Get()
-	if quit {
-		return false
-	}
-	defer c.queue.Done(dsKey)
-
-	err := c.sync()
-	if err == nil {
-		c.queue.Forget(dsKey)
-		return true
-	}
-
-	utilruntime.HandleError(fmt.Errorf("%v failed with : %v", dsKey, err))
-	c.queue.AddRateLimited(dsKey)
-
-	return true
+		secretClient:         v1helpers.CachedSecretGetter(kubeClient.CoreV1(), kubeInformers),
+	}
+	return factory.New().ResyncEvery(time.Minute).WithInformers(
+		kubeInformers.InformersFor("").Core().V1().Nodes().Informer(),
+		kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Secrets().Informer(),
+		kubeInformers.InformersFor(operatorclient.GlobalUserSpecifiedConfigNamespace).Core().V1().Secrets().Informer(),
+		infrastructureInformer.Informer(),
+		operatorClient.Informer(),
```

Comment:
we added  `operatorClient.Informer()` just making sure this was intended.

Reply:
it was weird that we waited for a cache sync in this informer but we have not added event handler to it... i think we want to react when the config CR change?